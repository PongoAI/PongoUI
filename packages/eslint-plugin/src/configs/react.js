/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
