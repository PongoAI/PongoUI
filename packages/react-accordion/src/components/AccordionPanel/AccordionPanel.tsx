import { AccordionPanel as FluentUIAccordionPanel } from '@fluentui/react-accordion';

/**
 * The AccordionPanel component is used to show and hide nested information.
 */
export const AccordionPanel = FluentUIAccordionPanel;
