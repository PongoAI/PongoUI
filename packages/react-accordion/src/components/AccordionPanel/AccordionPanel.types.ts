import type {
  AccordionPanelProps as FluentAccordionPanelProps,
  AccordionPanelState as FluentAccordionPanelState,
  AccordionPanelSlots as FluentAccordionPanelSlots,
} from '@fluentui/react-accordion';

export type AccordionPanelSlots = FluentAccordionPanelSlots;

export type AccordionPanelProps = FluentAccordionPanelProps;

export type AccordionPanelState = FluentAccordionPanelState;
