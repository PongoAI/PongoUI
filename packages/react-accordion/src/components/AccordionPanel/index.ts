export * from './AccordionPanel';
export * from './AccordionPanel.types';
