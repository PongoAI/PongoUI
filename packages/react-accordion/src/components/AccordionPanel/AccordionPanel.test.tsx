import * as React from 'react';
import { AccordionPanel } from './AccordionPanel';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';

describe('AccordionPanel', () => {
  isConformant({
    Component: AccordionPanel,
    displayName: 'AccordionPanel',
    disabledTests: ['component-handles-classname', 'component-handles-ref', 'component-has-root-ref'],
  });

  describe('Snapshot tests', () => {
    it('renders a basic AccordionPanel', () => {
      const result = render(<AccordionPanel />);

      expect(result.baseElement).toMatchSnapshot();
    });
  });
});
