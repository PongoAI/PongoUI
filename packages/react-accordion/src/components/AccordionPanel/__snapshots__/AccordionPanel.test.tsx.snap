// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionPanel Snapshot tests renders a basic AccordionPanel 1`] = `
<body>
  <div />
</body>
`;
